### PR TITLE
Dashboard - User Login

### PR DESCRIPTION
**Pull request by RidwanHamid501**
Complete user login side of dashboard. Addresses issue #36

### CODE REVIEW

File: dashboard/dashboard.py [hunk 1]
```diff
@@ -6,67 +6,68 @@
...
 from psycopg2.extras import RealDictCursor
+from email_validator import validate_email, EmailNotValidError
 
 load_dotenv()
 
 API_ENDPOINT = ENV["API_ENDPOINT"]
 
 
-def get_connection() -> psycopg2.extensions.connection:
+def get_connection() -> tuple:
```
Comment (SurinaCS) [inline dashboard/dashboard.py:16]: great type hints

File: dashboard/dashboard.py [hunk 2]
```diff
@@ -93,41 +91,45 @@ def execute_query(query: str, params: tuple = None, fetch_one: bool = False):
...
-    """Check if the given phone number is a valid UK mobile number"""
-    pattern = r'^07\d{9}$'
-    return bool(re.match(pattern, phone_number))
+def is_valid_email(email: str) -> bool:
+    """Check if entered email is valid"""
+    try:
+        validate_email(email)
+        return True
+    except EmailNotValidError:
+        return False
```
Comment (SurinaCS) [inline dashboard/dashboard.py:100]: nice

File: dashboard/dashboard.py [hunk 3]
```diff
@@ -137,122 +139,178 @@ def submit_topic(data: dict) -> None:
...
-        if user_first.strip() and user_last.strip() and phone_number.strip():
-            phone_number = phone_number.strip()
-            user_first = user_first.strip()
-            user_last = user_last.strip()
 
+def process_user(user_first: str, user_last: str, email: str) -> None:
+    """Processes the user's verification or registration"""
+    if not user_first or not user_last or not email:
+        st.session_state["verification_warning"] = "Please enter both your name and email."
+        st.session_state["user_verified"] = False
```
Comment (SurinaCS) [inline dashboard/dashboard.py:159]: nice

**Review by SurinaCS — APPROVED** [review]
good work!

File: dashboard/dashboard.py [hunk 4]
```diff
@@ -93,41 +91,45 @@ def execute_query(query: str, params: tuple = None, fetch_one: bool = False):
...
+def insert_user(first_name: str, last_name: str, email: str) -> None:
     """Insert a new user into the database"""
-    query = """INSERT INTO "user" (first_name, last_name, phone_number)
+    query = """INSERT INTO "user" (first_name, last_name, email)
                VALUES (%s, %s, %s);"""
-    execute_query(query, (first_name, last_name, phone_number))
+    execute_query(query, (first_name, last_name, email))
 
 
 def submit_topic(data: dict) -> None:
```
Comment (Kurt812) [inline dashboard/dashboard.py:129]: Could add some more error handling here

```suggestion
def submit_topic(data: dict) -> None:  # Modify this function
    try:
        response = requests.post(API_ENDPOINT, json=data, timeout=1000)
        response.raise_for_status()  # Add this for better error handling
        st.success("✅ Topic submitted successfully!")
    except requests.exceptions.HTTPError as http_err:
        st.error(f"HTTP error occurred: {http_err}")
    except requests.exceptions.RequestException as req_err:
        st.error(f"Failed to connect to the API. Error: {req_err}")
```

**Pull request merged**